src/components/register-page/register-page.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react'
+import styles from './register-page.module.css'
+import { Link } from 'react-router-dom'
+import { useSelector, useDispatch } from 'react-redux'
+import { register, REGISTER_FAILED } from '../../services/actions/user-actions'
+
+import InputLoader from '../input-loader/input-loader'
+import Message from '../message/message'
+
+import { Input, PasswordInput, Button } from '@ya.praktikum/react-developer-burger-ui-components'
+
+const RegisterPage = () => {
+  const [name, setName] = useState('')
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+
+  const dispatch = useDispatch()
+  const { registerRequest, registerFailed, registerMessage } = useSelector(store => store.user)
+  
+  const onChangeName = e => {
+    setName(e.target.value)
+  }
+  const onChangeEmail = e => {
+    setEmail(e.target.value)
+  }
+  const onChangePassword = e => {
+    setPassword(e.target.value)
+  }
+  const onSubmit = e => {
+    e.preventDefault()
+    if (password.length <= 3) {
+      dispatch({
+        type: REGISTER_FAILED,
+        payload: 'Пароль должен быть больше 3-х символов'
+      })
+    } else {
+      dispatch(register({
+        email: email,
+        password: password,
+        name: name
+      }))
+    }
+  }
+
+  return (
+    <div className={styles.box}>
+      <p className={`text text_type_main-medium ${styles.title}`}>Регистрация</p>
+      <form className={styles.form} onSubmit={e => onSubmit(e)}>
```

Comment:
Если для вызова функции нужен аргумент, который доступен при вызове функции, то можно прописать только функцию без вызова, аргумент автоматически передастся в функцию:
```suggestion
      <form className={styles.form} onSubmit={onSubmit}>
```